presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -312,4 +322,19 @@ private void enforceCpuLimits()
             }
         }
     }
+
+    /**
+     * Enforce query scan physical bytes limits
+     */
+    private void enforceScanLimits()
+    {
+        for (QueryExecution query : queryTracker.getAllQueries()) {
+            DataSize limit = getQueryMaxScanPhysicalBytes(query.getSession());
```

Comment:
Over here we are just using the session property - which a user can override. Similar to how this is being done for `enforceCpuLimits` where in we take the min of the cluster setting and session property and enforce based on that.

Reply:
right that's the idea, getQueryMaxScanPhysicalBytes obtains the value set from system property so that it can be compared with the actual scanned value

Reply:
Properties defined in `SystemSessionProperties` are properties that can be overridden by users by saying:

```
SET SESSION query_max_scan_physical_bytes=100000000000000000;
SELECT * FROM very_large_table;
```

If as a system administrator I want to cap the amountt of data a query can read - then this doesn't work since any user can override it.

If you check `enforceCpuLimit`, there we do:

```
Duration sessionLimit = getQueryMaxCpuTime(query.getSession()); // value from SystemSessionProperties
Duration limit = Ordering.natural().min(maxQueryCpuTime, sessionLimit);  // maxQueryCpuTime is from QueryManagerConfig
```

And hence in this example - a user can lower their allowed limit, but not go beyond what the system administrator has set as the cluster setting.